ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -716,19 +718,18 @@ private static String getContainerJavaOpts(Configuration conf) {
     }
     logLevel = sb.toString();
 
+    String finalOpts = null;
     if (HiveConf.getIntVar(conf, HiveConf.ConfVars.HIVETEZCONTAINERSIZE) > 0) {
-      if (javaOpts != null) {
-        return javaOpts + " " + logLevel;
-      } else  {
-        return logLevel;
-      }
+      finalOpts = javaOpts != null ? javaOpts + " " + logLevel : logLevel;
     } else {
       if (javaOpts != null && !javaOpts.isEmpty()) {
         LOG.warn(HiveConf.ConfVars.HIVETEZJAVAOPTS + " will be ignored because "
-                 + HiveConf.ConfVars.HIVETEZCONTAINERSIZE + " is not set!");
+            + HiveConf.ConfVars.HIVETEZCONTAINERSIZE + " is not set!");
       }
-      return logLevel + " " + MRHelpers.getJavaOptsForMRMapper(conf);
+      finalOpts = logLevel + " " + MRHelpers.getJavaOptsForMRMapper(conf);
     }
+    LOG.debug("Tez container log level: '{}', java opts: '{}', final opts: {}", logLevel, javaOpts, finalOpts);
```

Comment:
Do we need to log the values separately? Isn't finalOpts sufficient?

nit: in the log message finalOpts is not enclosed in single quotes as it happens for the rest.

Reply:
final opts is a result of some minor magic here above, so if it's not obvious why it is what it is, the values of the passed-in logLevel and javaOpts variables might help

Reply:
i don't see how`logLevel` & `javaOpts` could be useful. 
finalOpts always contains `logLevel` and depending on HIVETEZCONTAINERSIZE - `javaOpts`

Reply:
@deniskuzZ , right, I believe we know hive.tez.container.size from different log lines, so finalOpts contains everything we need (as we can figure out javaOpts/logLevel from that easily), so I'm fine with making this log message simpler, as you and @zabetak suggested